src/morx.rs

```diff
@@ -0,0 +1,1824 @@
+                    liga_subst.next_state = 0;
+                    liga_subst.component_stack.clear();
+                    liga_subst.process_glyphs(ligature_subtable)?;
+                }
+            }
+        }
+    }
+
+    // println!("The glyphs array after ligature substitutions: {:?}", glyphs);
+
+    Ok(())
+}
+
+//---------------------------------  Morx Substitution Test --------------------------------------------
+pub fn morx_substitution_test<'a>(scope: ReadScope<'a>) -> Result<(), ParseError> {
+    let morx_table = scope.read::<MorxTable<'a>>()?;
+
+    let glyph1: RawGlyph<()> = RawGlyph {
+        unicodes: tiny_vec![[char; 1]],
+        glyph_index: 3,
+        liga_component_pos: 0,
+        glyph_origin: GlyphOrigin::Direct,
+        small_caps: false,
+        multi_subst_dup: false,
+        is_vert_alt: false,
+        fake_bold: false,
+        fake_italic: false,
+        extra_data: (),
+        variation: None,
+    };
+
+    let glyph2: RawGlyph<()> = RawGlyph {
+        unicodes: tiny_vec![[char; 1]],
+        glyph_index: 604,
+        liga_component_pos: 0,
+        glyph_origin: GlyphOrigin::Direct,
+        small_caps: false,
+        multi_subst_dup: false,
+        is_vert_alt: false,
+        fake_bold: false,
+        fake_italic: false,
+        extra_data: (),
+        variation: None,
+    };
+
+    let glyph3: RawGlyph<()> = RawGlyph {
+        unicodes: tiny_vec![[char; 1]],
+        glyph_index: 547,
+        liga_component_pos: 0,
+        glyph_origin: GlyphOrigin::Direct,
+        small_caps: false,
+        multi_subst_dup: false,
+        is_vert_alt: false,
+        fake_bold: false,
+        fake_italic: false,
+        extra_data: (),
+        variation: None,
+    };
+
+    let glyph4: RawGlyph<()> = RawGlyph {
+        unicodes: tiny_vec![[char; 1]],
+        glyph_index: 528,
+        liga_component_pos: 0,
+        glyph_origin: GlyphOrigin::Direct,
+        small_caps: false,
+        multi_subst_dup: false,
+        is_vert_alt: false,
+        fake_bold: false,
+        fake_italic: false,
+        extra_data: (),
+        variation: None,
+    };
+
+    let glyph5: RawGlyph<()> = RawGlyph {
+        unicodes: tiny_vec![[char; 1]],
+        glyph_index: 3,
+        liga_component_pos: 0,
+        glyph_origin: GlyphOrigin::Direct,
+        small_caps: false,
+        multi_subst_dup: false,
+        is_vert_alt: false,
+        fake_bold: false,
+        fake_italic: false,
+        extra_data: (),
+        variation: None,
+    };
+
+    let mut glyphs: Vec<RawGlyph<()>> = vec![glyph1, glyph2, glyph3, glyph4, glyph5];
+
+    let features = Features::Custom(Vec::new());
+
+    let _res = apply(&morx_table, &mut glyphs, &features);
+
+    //println!("The glyphs array after morx substitutions: {:?}", glyphs);
+
+    //print glyph array after applying substitutions.
+    for glyph in glyphs.iter() {
+        println!("  {:?}", glyph);
+    }
```

Comment:
I think this should also be replaced by assertions that the glyphs are as expected.